app/services/trend_importer.rb

```diff
@@ -32,17 +32,34 @@ def update_trend_counters
       end
     end
     Trend.has_value.each do |trend|
+      new_count = if trend.is_a? FactTrend
+                    counter_hash[trend.trendable_id][trend.fact_value]
+                  else
+                    counter_hash[trend.trendable_type][trend.trendable_id]
+                  end || 0
+
+      latest_counter = trend.trend_counters.order(:created_at).last
+      if latest_counter
+        latest_counter.interval_end = timestamp
```

Comment:
Last measurement value is stored in `update_date` of the latest counter. If the value is changed, that means that we need to end the latest stored counter (set its `interval_end`), and begin a new interval. It is performed after the `next unless...` line.